### PR TITLE
Update gx1 initial condition, Icepack, tests, and version.  Fix hmix default value

### LINKED ISSUE #573: Add new landfast tests
See #565.  We need to update the landfast tests to test multiple options and test at a time where there is some landfast ice (April).  This should be done after the new time manager is merged and after we have the new ics from the long JRA55 spinup runs.

### LINKED ISSUE #585: CESM ocean forcing - hmix division by zero
In icepack_ocean.F90 line 103.
   sst = sst + dt * ( &
            (fsens_ocn + flat_ocn + flwout_ocn + flw + swabs) * (c1-aice) &
          + fhocn + fswthru)         &  ! these are *aice
          / (cprho*hmix)
hmix is zero when using standard ocean forcing (CESM) in standalone simulation.
This is initialized beginning line 4394 in ice_forcing.F90
Below is a hard coded fix (hmix=c20) instead of the value from the forcing file.
        do j = 1, ny_block
          do i = 1, nx_block
            if (n == 2) sss    (i,j,:) = c0
            if (n == 3) hmix   (i,j,:) = c0
            if (n == 4) uocn   (i,j,:) = c0
            if (n == 5) vocn   (i,j,:) = c0
            if (n == 6) ss_tltx(i,j,:) = c0
            if (n == 7) ss_tlty(i,j,:) = c0
            if (n == 8) qdp    (i,j,:) = c0
            do iblk = 1, nblocks
              if (hm(i,j,iblk) == c1) then
                if (n == 2) sss    (i,j,iblk) = work1(i,j,iblk)
                if (n == 3) hmix   (i,j,iblk) = c20  !!work1(i,j,iblk)
                if (n == 4) uocn   (i,j,iblk) = work1(i,j,iblk)
                if (n == 5) vocn   (i,j,iblk) = work1(i,j,iblk)
                if (n == 6) ss_tltx(i,j,iblk) = work1(i,j,iblk)
                if (n == 7) ss_tlty(i,j,iblk) = work1(i,j,iblk)
                if (n == 8) qdp    (i,j,iblk) = work1(i,j,iblk)
              endif
            enddo
          enddo
        enddo
      enddo

### LINKED ISSUE #588: Robustify GitHub Actions workflow
Sometimes the GitHub Actions CI runner encounters a broken connection to Zenodo:
https://github.com/CICE-Consortium/CICE/pull/587/checks?check_run_id=2230759756#step:11:58435
~~~
2021-03-30T19:22:27.0277120Z 2021-03-30 19:22:26 (4.69 MB/s) - Connection closed at byte 2537720689. Retrying.
2021-03-30T19:22:27.0277510Z 
2021-03-30T19:22:27.8913650Z --2021-03-30 19:22:27--  (try: 2)  https://zenodo.org/record/3728364/files/CICE_data_gx3_forcing_JRA55-20200320.tar.gz
2021-03-30T19:22:28.0625210Z Connecting to zenodo.org (zenodo.org)|137.138.76.77|:443... connected.
2021-03-30T19:22:28.5772140Z HTTP request sent, awaiting response... 429 Too Many Requests
2021-03-30T19:22:28.5773870Z 2021-03-30 19:22:28 ERROR 429: Too Many Requests.
~~~

The 'Download input data' step is successful in the workflow, despite wget encoutering an error. It's not the first time this happens, and I think I understand why. Each step in the workflow is ran by the shell specified by the "shell" keyword, or a platform default:
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell

We use the Cshell ("csh") (which in theory is not necessary, see #557) and thus use a custom shell:
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#custom-shell
https://github.com/CICE-Consortium/CICE/blob/e204fb826412288f8f789b2df3a71c9c78af2099/.github/workflows/test-cice.yml#L18-L20

but this invocation of csh does *not* ensure that any error encountered terminates the process, which is what is done by default for ex. Bash (see the Github documentation). So we should at least be using `/bin/csh -e {0}` to ensure errors are caught.
Also, we should invoke 'wget' with the `--progress=dot:giga` option to avoid countless number of lines in the CI log since the input data is really big.

_Originally posted by @phil-blain in https://github.com/CICE-Consortium/CICE/issues/587#issuecomment-810555119_

### LINKED ISSUE #589: Hourly output broken by recent time manager update
I've just noticed that hourly outputs does not work anymore since the update to the time manager. 

I'm comparing b720380 (Update Time Manager (#566), 2021-03-16) and 5a0a559 (Modified k1 value and some of the doc (#574), 2021-03-11) (the commit just before).

With these modification to the namelist:
~~~diff
diff --git 1/../before-time-manager-update2/ice_in 2/./ice_in
index 1dfb0b6..684c8ed 100644
--- 1/../before-time-manager-update2/ice_in
+++ 2/./ice_in
@@ -34,7 +33,7 @@ &setup_nml
     latpnt(2)      = -65.
     lonpnt(2)      = -45.
     dbug           = .false.
-    histfreq       = 'm','x','x','x','x'
+    histfreq       = 'h','x','x','x','x'
     histfreq_n     =  1 , 1 , 1 , 1 , 1
     hist_avg       = .true.
     history_dir    = './history/'
@@ -413,9 +412,9 @@ &icefields_nml
     f_VGRDb        = .false.
     f_VGRDa        = .true.
     f_bounds       = .false.
-    f_aice         = 'm' 
+    f_aice         = 'h' 
     f_hi           = 'm'
     f_hs           = 'm' 
~~~

With 5a0a559:
~~~
$ ll -rt $(cice_rundir)/history
total 12M
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_ic.2005-01-01-03600.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-32400.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-28800.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-25200.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-21600.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-18000.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-14400.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-10800.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-07200.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-75600.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-72000.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-68400.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-64800.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-61200.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-57600.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-54000.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-50400.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-46800.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-43200.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-39600.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-36000.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-02-00000.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-82800.nc
-rw-r--r-- 1 me mygroup 503K Apr  1 12:40 iceh_01h.2005-01-01-79200.nc
~~~

with b720380:
 ~~~
$ ll -rt $(cice_rundir)/history
total 504K
-rw-r--r-- 1 me mygroup 503K Apr  1 12:41 iceh_ic.2005-01-01-03600.nc
~~~

/cc @apcraig

### PR DESCRIPTION
**Pull request by apcraig**
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Update gx1 initial condition, Icepack, tests, and version number.  Fix hmix default value #585 
- [X] Developer(s): 
    apcraig, dabail10
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    tests are bit-for-bit except gx1 with new initial condition.  debug tests with gx1 fail due to Nans in ic file.  https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks#c0364f4f6924009f9d949d6c0484e832c1c5a9c6
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [X] Yes
    - [ ] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [X] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [X] No 
- [X] Please provide any additional information or relevant details below:

Also 
- Remove old and unused jra55_data subroutine, should have been done before.
- Fix default hmix value , closes #585
- Update landice tests, closes #573
- Update gx1prod settings to match up with recent gx1 jra55 multi-year run
- Update gx1 and gx3 initial filenames
- Update SOM forcing dataset name for gx1 runs
- Update CI wget implementation, closes #588
- Create variable in ice_forcing.F90 called mixed_layer_depth_default and use it instead of c20
- Fix bug in hourly output, created in time manager update , closes #589
- Set start year for all runs to 2005 and turn leap years on by default
- Update documentation in history output section to add information about hist_avg namelist, noted in #566
- Update color links on test results wiki page, move from https to local images
- #481, #249

### CODE REVIEW

File: cicecore/cicedynB/general/ice_forcing.F90
Comment (eclare108213) [inline cicecore/cicedynB/general/ice_forcing.F90:2673]: Just a note, mainly as a reminder:  We are making JRA the default, but the older forcing options are still available.  Those will need to be deprecated via our process (not in this release).

File: configuration/scripts/ice_in
Comment (eclare108213) [inline configuration/scripts/ice_in:4]: shouldn't year_init match the new forcing data?  maybe it doesn't matter, as long as it's properly set in the -s options.  I see below that fyear_init=2005, but I thought that one wasn't used now.  It's a little confusing, would be better to be consistent.

File: cicecore/cicedynB/general/ice_forcing.F90
Comment (eclare108213) [inline cicecore/cicedynB/general/ice_forcing.F90:4156]: Is this just temporary because of NaNs in the file?  If not, then it would be better to set this as a parameter at the top of the module, to make it more explicit.  'default mixed layer depth'

File: configuration/scripts/ice_in
Comment (apcraig) [reply configuration/scripts/ice_in:4]: I saw this and wondered the same things.  I was reluctant to make this change as I didn't understand the importance or history.  The other thing is that it will change all the gx3 results as well in our regression test.  But I agree 100% that it would be nice if all of this were consistent.  Are there other defaults that need to change.  Should use_leap_years be always true, for instance?  I can make this change as part of this PR, but it will change answers.

File: cicecore/cicedynB/general/ice_forcing.F90
Comment (apcraig) [reply cicecore/cicedynB/general/ice_forcing.F90:4156]: I think this avoids a divide by zero.  I assume there are also some science reasons for setting it to 20.  I can create a parameter in the code  as suggested and will do that later today.  I think that's a good idea.

**Comment by apcraig** [comment]
This is ready to merge.  I ran a full test suite on cheyenne this morning, https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks#4dc3bf15908deffbd68dd400fc0513a6bae7b81f and everything looks good.

It would be nice to merge this afternoon to leverage automated weekend testing and to carry out release testing this weekend as well.  Someone should review and approve.  I'm happy to make additional changes as needed too.

**Review by eclare108213 — APPROVED** [review]
Sorry I didn't notice this was waiting for a review earlier.  A few questions/comments to consider, but it looks good.

File: cicecore/cicedynB/general/ice_forcing.F90
Comment (apcraig) [reply cicecore/cicedynB/general/ice_forcing.F90:4159]: Maybe you're right.  I'm just implementing a more robust solution to avoid divide by zero.  I could undo the setting here, but would want to retest everything.  Let me know if you want me to do that.

File: configuration/scripts/options/set_nml.gx3sep2
Comment (apcraig) [reply configuration/scripts/options/set_nml.gx3sep2:3]: Yes it does.  It works just fine.  In fact, sec_init = 7200 as well in this setup in case you didn't notice and it all works fine.  It runs 5 days to 2005-09-07-07200.  The output is also as expected in the sense that things like daily history averages are fine except on the first and last day.  If one started on 2005-09-07-07200, you'd probably want to run 22 hours first (which you can do with npt_unit='h' and npt=22) to get on a daily boundary, and then start running days or months after that.

File: cicecore/cicedynB/general/ice_forcing.F90
Comment (eclare108213) [reply cicecore/cicedynB/general/ice_forcing.F90:4159]: We seem to be in a transition between initializing everything in sight (what I've always done), and letting compiler flags use uninitialized values to find problems.  In my opinion, if this line is initialized to anything, it ought to be zero.  That would similarly catch places in the code that are accessing land values (and shouldn't be).  However if there's still weirdness with the initialization on padded cells or halos (isn't that how this all got started?), then leave it like this.  It looks out of place to me but I can countenance the need for it.

File: configuration/scripts/options/set_nml.gx3sep2
Comment (eclare108213) [reply configuration/scripts/options/set_nml.gx3sep2:3]: cool, thanks

File: cicecore/cicedynB/general/ice_forcing.F90
Comment (apcraig) [reply cicecore/cicedynB/general/ice_forcing.F90:4159]: Let me change this back to zero and I'll retest.  If that works, I'll push and merge.  If not, I'll just leave it as is and merge.

**Review by dabail10 — APPROVED** [review]
This looks good to me. I do wonder about setting a mixed-layer depth default. Particularly as I did a fill, so there should be valid values everywhere.

File: cicecore/cicedynB/general/ice_forcing.F90
Comment (apcraig) [reply cicecore/cicedynB/general/ice_forcing.F90]: Setting this to c0 is fine and I have committed that and will merge as soon as the github actions testing passes.  If there are any other questions, feel free to raise them.  thanks.

File: cicecore/cicedynB/general/ice_forcing.F90
Comment (TillRasmussen) [reply cicecore/cicedynB/general/ice_forcing.F90]: This is a way of avoiding divison by zero.

**Pull request merged**
